src/main/java/org/sonarsource/auth/github/GitHubIdentityProvider.java

```diff
@@ -112,6 +113,11 @@ public void callback(CallbackContext context) {
     UserIdentity userIdentity = userIdentityFactory.create(getUser(scribe, accessToken),
       settings.syncGroups() ? getTeams(scribe, accessToken) : null);
 
+    if (isOrganizationMembershipRequired() && !isOrganizationsMember(accessToken, userIdentity.getLogin())) {
+      throw new IllegalStateException(format("'%s' must be a member of at least one organization: '%s'",
```

Comment:
Please don't throw an IllegalStateException => it's generating an exception on server side and the message is not displayed to the user.
You must use instead a `org.sonar.api.server.authentication.UnauthorizedException` : it will directly return the message to the user, and no error will be generated in server's log